src/portal_to_terra_input_from_anaset.py

```diff
@@ -0,0 +1,298 @@
+
+    # def get_data_access_auth(self):
+    #     seqfile_accessions = self.data['atac_read1_accessions'] + self.data['atac_read2_accessions'] + self.data['rna_read1_accessions'] + self.data['rna_read2_accessions']
+    #     file_controlled_access = 0
+    #     for seqfile_acc in seqfile_accessions:
+    #         seqfile_item = self.igvf_api.sequence_files(accession=[seqfile_acc]).graph[0]
+    #         file_controlled_access += seqfile_item.controlled_access
+    #     if file_controlled_access > 0:
+    #         self.data['controlled_access'] = True
+
+    def build_input_dict(self) -> dict:
+        """Build the final output report dict
+        """
+        self.data['analysis_set_acc'] = ','.join(self.analysis_set_acc)     # Technically it is a list of one
+        # self.data['lab'] = self.lab
+        # self.data['awards'] = self.awards
+        self.data['taxa'] = self.samp_taxa
+        # Get seqfile URLs
+        self.get_seqfile_uris_and_seqspec_by_assays_and_runs()
+        # Make unique seqspec list
+        self.data['seqspec_accessions'] = list(self.data['seqspec_accessions'])
+        self.data['seqspec_urls'] = list(self.data['seqspec_urls'])
+        return self.data
+
+
+def main(query_analysis_set_accs: list, igvf_api, save_to_file: bool = False, output_path='./') -> pd.DataFrame:
+    """Full script to go from a query measurement set accession to a uniform pipeline input TSV.
+
+    Args:
+        query_measet_accs (list): A list of MeaSet accessions for the uniform pipeline
+        igvf_api (_type_): _description_
+        save_to_file (bool, optional): Defaults to pipeline_input_table.
+        output_path (str, optional): Defaults to './'.
+
+    Returns:
+        pd.DataFrame: Final uniform pipeline input tsv.
+    """
+    pipeline_input_list = []
+    for anaset_acc in query_analysis_set_accs:
+        # curr_samp_id = get_samp_ids_from_measets(measurement_set_accessions=[measet_acc], igvf_api=igvf_api)
+        curr_input_builder = SingleCellInputBuilder(analysis_set_acc=[anaset_acc], igvf_api=igvf_api)
+        pipeline_input_list.append(curr_input_builder.build_input_dict())
+    pipeline_input_table = pd.DataFrame(pipeline_input_list)
+    # Output TSV if needed
+    if save_to_file:
+        curr_datetime = datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
+        pipeline_input_table.to_csv(os.path.join(output_path, f'single-cell_uniform_pipeline_input_table_{curr_datetime}.tsv'), sep="\t")
+    else:
+        return pipeline_input_table
+
+
+if __name__ == '__main__':
+    # Tester measurement set
+    nonmultiplexed_measets = ['IGVFDS0020LQRY', 'IGVFDS7133CHRI']
+    multiplexed_measets = ['IGVFDS5205JRNX', 'IGVFDS3910GBDQ']
+    analysis_sets = ['IGVFDS0006MFFD', 'IGVFDS9202ZPBI']
+    main(query_analysis_set_accs=analysis_sets,
+         igvf_api=get_igvf_auth_and_api(),
+         save_to_file=True,
+         output_path='/Users/zheweishen/IGVF/single_cell_fg/Portal_input_output/test_input'
```

Comment:
This default output_path should be a constant. And this path here may not work for other people. Can use relative path, for example: OUTPUT_FOLDER = './single_cell_input'